frontend/src/apps/qotd/containers/Result.js

```diff
@@ -10,35 +10,27 @@ import Unavailable from '../components/Unavailable'
 
 import { getQotd } from '../actions'
 
-class Answer extends React.Component {
-  state = { fetchFailed: false }
+function Answer(props) {
+  const [fetchFailed, setFetchFailed] = useState(false)
 
-  componentDidMount () {
-    const { qotd } = this.props
+  useEffect(() => {
+    const { qotd } = props
     // Only fetch qotd if not already in redux or more than 24 hours old
     if (!qotd || Date.now() - new Date(qotd.day) > 24 * 60 * 60 * 1000) {
-      this.props.getQotd().catch(() => this.setState({ fetchFailed: true }))
+      props.getQotd().catch(() => setFetchFailed(true))
     }
-  }
+  }, [props])
```

Comment:
`const { qotd } = props` should be outside of the `useEffect` callback.

For `useEffect` to mimic `componentDidMount`, the dependency array must be empty `[]`.